mesonbuild/interpreter.py

```diff
@@ -703,9 +703,12 @@ def get_custom_install_mode(self):
         return self.custom_install_mode
 
 class DataHolder(InterpreterObject, ObjectHolder):
-    def __init__(self, data):
+    def __init__(self, data, subproject: str):
         InterpreterObject.__init__(self)
-        ObjectHolder.__init__(self, data)
+        ObjectHolder.__init__(self, data, subproject)
+        self.methods.update({
+            'sources': self.sources_method,
+        })
```

Comment:
Hmm, Data object is not documented. The only place it's currently used is the return value of install_data() that is documented as being void... I'm not against adding this method, but we need to add a section under "returned objects" of reference manual.

Reply:
The more I think about it, the more I think this should be a custom_target. We want it to be updated whenever any of the inputs are updated anyway.